### PR TITLE
Option to ignore "Reduce duplication" messages

### PR DESCRIPTION
**Pull request by wapxmas**

### CODE REVIEW

**Comment by wapxmas** [comment]
"Reduce duplication" messages are quite controversial, I propose to make the option that allows you to ignore them.

**Comment by Arcanemagus** [comment]
If they are controversial, that means that there are people that would _want_ them as well, can you put this behind a configuration defaulting to false (disabled)?

**Comment by wapxmas** [comment]
Yes it would matches the current behavior.

File: lib/init.coffee [hunk 1]
```diff
@@ -37,7 +37,7 @@ module.exports =
           hints = (('--hint=' + h) for h in @hlintHints)
           process = new BufferedProcess
             command: @executablePath
-            args: [filePath, '--json'].concat hints
+            args: [filePath, '--json', '--ignore=Reduce duplication'].concat hints
```
Comment (Arcanemagus) [inline lib/init.coffee]: Add an option controlling this, defaulting to not enabled.

File: lib/init.coffee [hunk 2]
```diff
@@ -12,6 +12,10 @@ module.exports =
       default: ['Default', 'Dollar', 'Generalise']
       items:
         type: 'string'
+    hlintIgnoreReduceDuplication:
```
Comment (Arcanemagus) [inline lib/init.coffee]: As this is already behind the `linter-hlint` tag, there is no need to prefix this with `hlint`. Can you rename this to `ignoreReduceDuplication`?

**Pull request merged**